ntc_templates/templates/juniper_junos_show_chassis_hardware.textfsm

```diff
@@ -0,0 +1,19 @@
+Value Filldown FPC (\d+)
+Value Filldown MIC (\d+)
+Value Filldown PIC (\d+)
+Value XCVR (\w+)
+Value PART (\S+)
+Value SERIAL (\S+)
+Value DESCRIPTION (\S+)
+
+Start
+  ^\s+Hardware\s+
+  ^.*FPC\s+${FPC} -> Continue
+  ^\s+CPU.*
+  ^.*MIC\s+${MIC}\s+\S+\s+\S+\s+\S+.* -> Continue
+  ^.*PIC\s+${PIC}\s+\S+\s+\S+\s+\S+.* -> Continue
+  ^.*Xcvr\s+${XCVR}\s+(REV\s+\d+|)\s+${PART}\s+${SERIAL}\s+${DESCRIPTION} -> Continue.Record
+  ^\s*$$
+  ^(Power|Fan)\s+
+
```

Comment:
This is missing the EOF as defined at the end of this section: https://github.com/networktocode/ntc-templates#states

Reply:
@jmcgill298 I just to clarify should this template and actually the other one (show interfaces) should like this:

```
^\s*$$
^(Power|Fan)\s+
^. -> Error
```
I didn't see the error line on the others and it passed the tests. So I had just matched to the others. Just let me know the above is correct and I will fix it.

Reply:
you will probably want to catch empty lines first.

```
...
  ^\s*$$
  ^. -> Error
```

This does require all lines to be matched (not necessarily captured), so if the template is ignoring some lines of output, it will need to be updated to account for those.

For the modifications of an existing template, we do not require it to be updated to have that line.

Reply:
hmm ok. so there is a lot of variation in the starting lines between the platforms (MX, QFX, SRX, etc..discovered from testing locally). It worked against the MX and QFX in its current state. If that is a requirement I will need to add additional raw outputs and tests from the other platforms. I will be using against those platforms and if I add that error it will break the capture for QFXs.

Reply:
that is a requirement, as it is the only way we can have any assurance that the data that is intended to be captured is actually being captured. Otherwise it will ignore the relevant data and report back a subset of what is desired, and the end user would not be aware they had incorrect data